lib/manager/argocd/extract.ts

```diff
@@ -0,0 +1,61 @@
+import { safeLoadAll } from 'js-yaml';
+import * as gitTags from '../../datasource/git-tags';
+import * as helm from '../../datasource/helm';
+import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
+import type { ApplicationDefinition } from './types';
+import { fileTestRegex } from './util';
+
+function loadYaml(content: string): ApplicationDefinition[] {
+  return safeLoadAll(content);
+}
+function createDependency(
+  definition: ApplicationDefinition
+): PackageDependency {
+  let result: PackageDependency;
+
+  const source = definition.spec.source;
```

Comment:
```suggestion
  const source = definition.spec?.source;
```

If there is a malformed document in file which is missing `spec`.